pennylane/estimator/resource_tracking.py

```diff
@@ -0,0 +1,379 @@
+        gate_set (set): the set of operations to track resources with respect to
+        scalar (int | None): optional scalar to multiply the counts. Defaults to 1.
+        config (dict | None): additional parameters to specify the resources from an operator. Defaults to ResourceConfig.
+    """
+    if gate_set is None:
+        gate_set = DefaultGateSet
+
+    if config is None:
+        config = ResourceConfig()
+
+    ## If op in gate_set add to resources
+    if cp_rep.name in gate_set:
+        gate_counts_dict[cp_rep] += scalar
+        return
+
+    ## Else decompose cp_rep using its resource decomp [cp_rep --> list[GateCounts]] and extract resources
+    decomp_func, kwargs = _get_decomposition(cp_rep, config)
+
+    params = {key: value for key, value in cp_rep.params.items() if value is not None}
+    filtered_kwargs = {key: value for key, value in kwargs.items() if key not in params}
+    resource_decomp = decomp_func(**params, **filtered_kwargs)
+    qubit_alloc_sum = _sum_allocated_wires(resource_decomp)
+
+    for action in resource_decomp:
+        if isinstance(action, GateCount):
+            _update_counts_from_compressed_res_op(
+                action.gate,
+                gate_counts_dict,
+                wire_mngr=wire_mngr,
+                scalar=scalar * action.count,
+                gate_set=gate_set,
+                config=config,
+            )
+            continue
+
+        if isinstance(action, Allocate):
+            if qubit_alloc_sum != 0 and scalar > 1:
+                wire_mngr.grab_zeroed(action.num_wires * scalar)
+            else:
+                wire_mngr.grab_zeroed(action.num_wires)
+        if isinstance(action, Deallocate):
+            if qubit_alloc_sum != 0 and scalar > 1:
+                wire_mngr.free_wires(action.num_wires * scalar)
+            else:
+                wire_mngr.free_wires(action.num_wires)
+
+    return
+
+
+def _sum_allocated_wires(decomp):
+    """Sum together the allocated and released wires in a decomposition."""
+    s = 0
+    for action in decomp:
+        if isinstance(action, Allocate):
+            s += action.num_wires
+        if isinstance(action, Deallocate):
+            s -= action.num_wires
+    return s
+
+
+@QueuingManager.stop_recording()
+def _ops_to_compressed_reps(
+    ops: Iterable[Operation | ResourceOperator],
+) -> list[CompressedResourceOp]:
+    """Convert the sequence of operations to a list of compressed resource ops.
+
+    Args:
+        ops (Iterable[Union[Operation, ResourceOperator]]): set of operations to convert
+
+    Returns:
+        List[CompressedResourceOp]: set of converted compressed resource ops
+    """
+    cmp_rep_ops = []
+    for op in ops:  # We are skipping measurement processes here.
+        if isinstance(op, ResourceOperator):
+            cmp_rep_ops.append(op.resource_rep_from_op())
+
+        if isinstance(op, Operation):  # map: op --> res_op, then: res_op --> cmprsd_res_op
```

Comment:
Are these comments necessary or just left by mistake? If later, please remove them.

Reply:
Not sure. Jay left these comments but they seem somewhat obvious to me? I can remove them.

Reply:
If seem obvious, please remove.